subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -2051,20 +2051,17 @@ uint16_t ull_conn_lll_max_tx_octets_get(struct lll_conn *lll)
 
 /**
  * @brief Initialize pdu_data members that are read only in lower link layer.
+ *        Fields that are modified after static initialization are only
+ *        re-initialized.
  *
  * @param pdu_tx Pointer to pdu_data object to be initialized
  */
 void ull_pdu_data_init(struct pdu_data *pdu_tx)
 {
-	LL_ASSERT(pdu_tx);
-
-	pdu_tx->cp = false;
-	pdu_tx->rfu = 0U;
```

Comment:
I think this could be done here instead of LLL. This member should be assigned in LLL only for fragmented PDUs more precisely for fragments that follow first one.

Reply:
As these fields will be set during the Tx PDU prepare LLL and can avoid a conditional set for fragmented PDUs, the assignments here is redundant CPU use.

Reply:
It seems to me that there is no need for additional conditions in LLL. The `cp`  and `resv` are already set for PDU fragments that follow the first one PDU: https://github.com/zephyrproject-rtos/zephyr/blob/59a8b19eaf5c653ddae564c39acec77be4d75ebe/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn.c#L657-L665. 
You can move:  https://github.com/zephyrproject-rtos/zephyr/blob/59a8b19eaf5c653ddae564c39acec77be4d75ebe/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn.c#L682 after this line https://github.com/zephyrproject-rtos/zephyr/blob/59a8b19eaf5c653ddae564c39acec77be4d75ebe/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn.c#L665 and remove https://github.com/zephyrproject-rtos/zephyr/blob/59a8b19eaf5c653ddae564c39acec77be4d75ebe/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn.c#L684-L688.

Then if you revert back removed lines from this function everything should be working. Also less code would be executed in ISR context.

Correct me if I'm wrong, please.

Reply:
> It seems to me that there is no need for additional conditions in LLL. The cp and resv are already set for PDU fragments that follow the first one PDU

No, `p = (void *)(tx->pdu + lll->packet_tx_head_offset);` makes `p` point to last 3 bytes of the previous fragment hence the explicit assignment for each `PDU_DATA_LLID_DATA_CONTINUE` PDU is required.

Regarding moving `p->rfu = 0U;`  and removing `p->resv = 0U;`, they need to be where they are as these two are memory allocated from free node tx pool and uninitialized fields when `ll_tx_mem_enqueue` is called. They are uninitialized in HCI/ULL layer so that LLL can use them as needed/reserved  for future use (I assume CTE feature will set `cp` and `resv` union field late in the data path just before Radio h/w packet pointer is set). Such similar use may arise on future Bluetooth Specifications for `rfu` and newer `resv` fields added to PDU format.

Reply:
> No, p = (void *)(tx->pdu + lll->packet_tx_head_offset); makes p point to last 3 bytes of the previous fragment hence the explicit assignment for each PDU_DATA_LLID_DATA_CONTINUE PDU is required.

Correct, so the only required additional condition statement is already there. If CP bit and RFU for first PDU are initialized in ULL there is no need for additional code in LLL (except one for continuation PDUs, that is already there).

The only thing is that, are the CP and/or RFU bits updated after initialization in ULL for first PDU?

Reply:
The CP bit (and future any RFU) could be updated after initialization in thread/HCI, say CP bit has to be removed so that enqueued data packets be transmitted over coded PHY without CTE.

Reply:
Ok let it be.